contracts/src/Verifier.sol

```diff
@@ -73,101 +119,289 @@ contract Verifier is IVerifier, Ownable {
         _;
     }
 
-    // TODO: enable solhint
-    /* solhint-disable */
+    modifier checkIfCommitExists(Hash commitId) {
+        require(
+            Hash.unwrap(commitIdToCommit[commitId].id) != "",
+            "commit not found"
+        );
+        _;
+    }
 
-    // TODO: implement constructor
-    constructor() {}
+    modifier isValidProver(address proverAddress) {
+        require(proverAddress == msg.sender, "invalid prover");
+        _;
+    }
 
-    /* solhint-enable */
+    /*
+        constructor
+    */
+
+    constructor(uint8 _difficulty) isValidDifficulty(_difficulty) {
+        difficulty = _difficulty;
+    }
+
+    /*
+        functions
+    */
 
     function registerModel(
-        Hash modelContentId,
-        string calldata modelName,
-        string calldata modelDescription
+        Hash _modelContentId,
+        string calldata _modelName,
+        string calldata _modelDescription
```

Comment:
Modified function parameters to be consistent with naming convention (prefixed by `_`).